src/averageAges.js

```diff
@@ -15,11 +15,16 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people
+    .filter(man => man.sex === 'm');
+  const menOfCentury = men.filter(year =>
+    Math.ceil(year.died / 100) === century);
+
+  return century
+    ? menOfCentury.reduce((sumAges, age) =>
+      sumAges + (age.died - age.born) / menOfCentury.length, 0)
+    : men.reduce((sumAges, age) =>
+      sumAges + (age.died - age.born) / men.length, 0);
```

Comment:
these lines are almost identical, the only difference is the array they are run on. Determine this array with `century` variable before, and run this reduce method only once